lib/MusicBrainz/Server/Data/Series.pm

```diff
@@ -209,6 +235,22 @@ sub get_entities {
     });
 }
 
+sub load_entity_count {
+    my ($self, @series) = @_;
+    return unless @series;
+
+    for my $series (@series) {
+        my $entity_type = $series->type->item_entity_type;
+        my $query = <<~"SQL";
+            SELECT count(*)
+              FROM ${entity_type}_series es
+             WHERE es.series = ?
+            SQL
+        my $count = $self->sql->select_single_value($query, $series->id);
+        $series->entity_count($count);
+    }
```

Comment:
we only need to perform one query per entity type here (GROUP BY series)

Reply:
So first you would split series by their entity type, then for each type do a query like 

```
            SELECT series, count(*)
              FROM ${entity_type}_series
             WHERE series = any(?)
       GROUP BY series
```

And then what, find the right series based on ID and assign the entity count? That last part sounds like it would also be a bit slow but probably less than different queries - but maybe I'm missing an easier option :)

Reply:
Yes, you could even do it in one query using `UNION ALL`. It won't be slow: an in-memory hash lookup should be orders of magnitude faster than performing any kind of I/O. :)